client/src/main/java/io/split/service/SplitHttpClientKerberosImpl.java

```diff
@@ -0,0 +1,220 @@
+                getHttpURLConnection.setRequestProperty(HEADER_CACHE_CONTROL_NAME, HEADER_CACHE_CONTROL_VALUE);
+            }
+
+            _log.debug(String.format("Request Headers: %s", getHttpURLConnection.getRequestProperties()));
+
+            int responseCode = getHttpURLConnection.getResponseCode();
+
+            if (_log.isDebugEnabled()) {
+                _log.debug(String.format("[%s] %s. Status code: %s",
+                        getHttpURLConnection.getRequestMethod(),
+                        getHttpURLConnection.getURL().toString(),
+                        responseCode));
+            }
+
+            String statusMessage = "";
+            if (responseCode < HttpURLConnection.HTTP_OK || responseCode >= HttpURLConnection.HTTP_MULT_CHOICE) {
+                _log.warn(String.format("Response status was: %s. Reason: %s", responseCode,
+                        getHttpURLConnection.getResponseMessage()));
+                statusMessage = getHttpURLConnection.getResponseMessage();
+            }
+
+            inputStreamReader = new InputStreamReader(getHttpURLConnection.getInputStream());
+            BufferedReader br = new BufferedReader(inputStreamReader);
+            String strCurrentLine;
+            String responseBody = new String();
+            while ((strCurrentLine = br.readLine()) != null) {
+                responseBody = responseBody + strCurrentLine;
+            }
+            return new SplitHttpResponse(responseCode,
+                    statusMessage,
+                    responseBody,
+                    getResponseHeaders(getHttpURLConnection));
+        } catch (Exception e) {
+            throw new IllegalStateException(String.format("Problem in http get operation: %s", e), e);
+        } finally {
+            try {
+                if (inputStreamReader != null) {
+                    inputStreamReader.close();
+                }
+            } catch (Exception e) {
+                _log.error(String.format("Could not close HTTP Stream: %s", e), e);
+            }
+        }
+    }
+
+    public synchronized SplitHttpResponse post(URI uri, HttpEntity entity, Map<String, List<String>> additionalHeaders) throws IOException {
+        HttpURLConnection postHttpURLConnection = null;
+        try {
+            postHttpURLConnection = (HttpURLConnection) uri.toURL().openConnection();
+            return _post(postHttpURLConnection, entity, additionalHeaders);
+        } catch  (Exception e) {
+            throw new IllegalStateException(String.format("Problem in http post operation: %s", e), e);
+        } finally {
+            try {
+                if (postHttpURLConnection != null) {
+                    postHttpURLConnection.disconnect();
+                }
+            } catch (Exception e) {
+                _log.error(String.format("Could not close URL Connection: %s", e), e);
+            }
+        }
+    }
+
+    public SplitHttpResponse _post(HttpURLConnection postHttpURLConnection,
```

Comment:
```suggestion
    public SplitHttpResponse post(HttpURLConnection postHttpURLConnection,
```

Reply:
This is an internal call that takes HttpURLConnection to make the testing easier.
There is another get() call which takes URI as the first parameter